### ISSUE TITLE
i got error on build 

### ISSUE THREAD

**Issue by takegine** [lerna/lerna#3356]
# error message
```
0 silly argv {
0 silly argv   _: [ 'run' ],
0 silly argv   lernaVersion: '5.6.1',
0 silly argv   '$0': 'C:\\Users\\ACE\\AppData\\Roaming\\npm\\pnpm-global\\5\\node_modules\\lerna\\cli.js',
0 silly argv   script: 'build'
0 silly argv }
1 notice cli v5.6.1
2 verbose rootPath D:\Users\ace\Documents\gitee\tools
3 verbose packageConfigs Package manager 'pnpm' detected. Resolving packages using 'pnpm-workspace.yaml'.
4 info Executing command in 2 packages: "pnpm run build"
5 error TypeError: Cannot read properties of undefined (reading 'version')
5 error     at D:\Users\ace\Documents\gitee\tools\node_modules\.pnpm\@lerna+package-graph@5.6.1\node_modules\@lerna\package-graph\index.js:80:37
5 error     at Array.forEach (<anonymous>)
5 error     at D:\Users\ace\Documents\gitee\tools\node_modules\.pnpm\@lerna+package-graph@5.6.1\node_modules\@lerna\package-graph\index.js:60:38
5 error     at Map.forEach (<anonymous>)
5 error     at new PackageGraph (D:\Users\ace\Documents\gitee\tools\node_modules\.pnpm\@lerna+package-graph@5.6.1\node_modules\@lerna\package-graph\index.js:49:10)
5 error     at new QueryGraph (D:\Users\ace\Documents\gitee\tools\node_modules\.pnpm\@lerna+query-graph@5.6.1\node_modules\@lerna\query-graph\query-graph.js:48:18)
5 error     at runTopologically (D:\Users\ace\Documents\gitee\tools\node_modules\.pnpm\@lerna+run-topologically@5.6.1\node_modules\@lerna\run-topologically\run-topologically.js:23:17)
5 error     at RunCommand.runScriptInPackagesTopological (D:\Users\ace\Documents\gitee\tools\node_modules\.pnpm\@lerna+run@5.6.1\node_modules\@lerna\run\index.js:162:17)

```

# lerna.json
```json
{
  "version": "0.0.1",
  "useWorkspaces": true,
  "npmClient": "pnpm"
}

```

when i run `lerna run build` i got the error. i have tried version 5.5.2 & 5.6.1.
but i can `cd` to packages/* then run `pnpm build`, that is success whitout error message. so what's the problem?

**Comment by takegine** [comment]
I solved this problem, created a new folder, and then `lerna init`, and then copy `packages/**` to new addon.

**Comment by takegine** [comment]
i use `lerna create panorama-cli` to add new package. and then I code it. `pnpm i` & `pnpm build` is success. but it fail when `lerna publish`
![image](https://user-images.githubusercontent.com/64788603/194763200-5996ecd4-3549-4594-b440-90ead930aba2.png)
![image](https://user-images.githubusercontent.com/64788603/194763068-6658cded-ae6e-4df2-8897-7f041c8c12ea.png)
```json
38 error TypeError: Cannot read properties of undefined (reading 'version')
38 error     at D:\Users\ace\Documents\gitee\tools\node_modules\.pnpm\@lerna+package-graph@5.6.1\node_modules\@lerna\package-graph\index.js:80:37
38 error     at Array.forEach (<anonymous>)
38 error     at D:\Users\ace\Documents\gitee\tools\node_modules\.pnpm\@lerna+package-graph@5.6.1\node_modules\@lerna\package-graph\index.js:60:38
38 error     at Map.forEach (<anonymous>)
38 error     at new PackageGraph (D:\Users\ace\Documents\gitee\tools\node_modules\.pnpm\@lerna+package-graph@5.6.1\node_modules\@lerna\package-graph\index.js:49:10)
38 error     at new QueryGraph (D:\Users\ace\Documents\gitee\tools\node_modules\.pnpm\@lerna+query-graph@5.6.1\node_modules\@lerna\query-graph\query-graph.js:48:18)
38 error     at runTopologically (D:\Users\ace\Documents\gitee\tools\node_modules\.pnpm\@lerna+run-topologically@5.6.1\node_modules\@lerna\run-topologically\run-topologically.js:23:17)
38 error     at D:\Users\ace\Documents\gitee\tools\node_modules\.pnpm\@lerna+version@5.6.1_nx@14.8.3+typescript@4.8.4\node_modules\@lerna\version\index.js:579:7
38 error     at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

**Comment by fahslaj** [comment]
Hi @takegine , thank you for opening an issue. This appears to be the same problem as #3343 . I am currently working on a fix with PR #3358 .

**Comment by takegine** [comment]
> Hi @takegine , thank you for opening an issue. This appears to be the same problem as #3343 . I am currently working on a fix with PR #3358 .

Hi,thank for your return. I will follow #3343
